nemo/utils/exp_manager.py

```diff
@@ -18,84 +18,218 @@
     """
     exp_manager is a helper function used to manage folders for experiments. It follows the pytorch lightning paradigm
     of root_dir/model_or_experiment_name/version. If the lightning trainer has a logger, exp_manager will get root_dir,
     name, and version from the logger. Otherwise it will use the root_dir and name arguments to create the logging
-    directory. The version will be a datetime string if running single node, and version will be an integer if running
-    on slurm multi-node.
-    It optionally creates TensorBoardLogger, and ModelCheckpoint objects from pytorch lightning. It copies sys.argv,
-    and git information if available to the logging directory. It creates a log file for each process to log their
-    output into.
+    directory. exp_manager also allows for explicit folder creation via explicit_log_dir.
+    The version will be a datetime string if running single node, and version will be an integer if running
+    on slurm multi-node. Datestime version can be disabled if use_datetime_version is set to False.
+    It optionally creates TensorBoardLogger, WandBLogger, ModelCheckpoint objects from pytorch lightning. It copies
+    sys.argv, and git information if available to the logging directory. It creates a log file for each process to log
+    their output into.
+    exp_manager additionally has a resume feature which can be used to continuing training from a previous_log_dir.
 
     Args:
         trainer (pytorch_lightning.Trainer): The lightning trainer.
         cfg (DictConfig, dict): Can have the following keys:
-            - name (str): The name of the experiment. Required argument.
+            - explicit_log_dir (str, Path): Can be used to override root_dir/name/version folder creation. Defaults to
+                None, which will use root_dir, name, and version to construct the logging directory.
             - root_dir (str, Path): The base directory to create the logging directory. Defaults to None, which logs to
                 ./NeMo_experiments.
+            - name (str): The name of the experiment. Defaults to None which turns into "default" via name = name or
+                "default".
+            - version (str): The version of the experiment. Defaults to None which uses either a datetime string or
+                lightning's TensorboardLogger system of using version_{int}.
+            - use_datetime_version (bool): Whether to use a datetime string for version. Defaults to True.
+            - resume (bool): Whether this experiment is resuming from a previous run. If True, previous_log_dir must
+                be set. If True, it sets trainer.resume_from_checkpoint so that the trainer should auto-resume.
+                exp_manager will move files under previous_log_dir to previous_log_dir/run_{int}.Defaults to False.
+            - previous_log_dir (bool): Used when resume = True to find previous checkpoints under
```

Comment:
Previous log dir here is marked as a bool, but later in code is assumed to be a path? Can we use bool and exp_manager auto resolve the path?

Reply:
removed previous log dir